### PR TITLE
Prototype Atom Output Ports, UI notifications

### PR DESCRIPTION
**Pull request by x42**
Updated PR #48 

Rebased on master (was temposync2), tested with https://github.com/x42/modspectre.lv2
with most-host/atom-send (d4e51e29d4) rebased onto current mod-host/master (2f1b067)

### CODE REVIEW

File: html/js/host.js [hunk 1]
```diff
@@ -92,6 +92,15 @@ $('document').ready(function() {
             return
         }
 
+        if (cmd == "output_atom") {
```
Comment (falkTX) [inline html/js/host.js]: not sure why this is still here.
it's already part of the latest code.
https://github.com/moddevices/mod-ui/blob/master/html/js/host.js#L104

File: mod/host.py [hunk 2]
```diff
@@ -1152,7 +1152,10 @@ def report_current_state(self, websocket):
             for symbol, value in pluginData['outputs'].items():
                 if value is None:
                     continue
-                websocket.write_message("output_set %s %s %f" % (pluginData['instance'], symbol, value))
+                if isinstance (value, float):
+                    websocket.write_message("output_set %s %s %f" % (pluginData['instance'], symbol, value))
+                if isinstance (value, str):
```
Comment (falkTX) [inline mod/host.py]: as I mentioned on IRC before, I don't agree with this.
note https://github.com/moddevices/mod-ui/blob/master/mod/host.py#L829
the last message of output-atom from mod-host is not saved. we should be saving it per port *and* key, not a single atom message per port.
also the current code is not saving that last value anyway, so this specific code is never reached

File: html/js/host.js (hunk 1 above)
Comment (x42) [inline html/js/host.js]: good catch. `git rebase` screwed this up for me.

File: mod/host.py (hunk 2 above)
Comment (x42) [inline mod/host.py]: Why save the message at all?
And it is currently reached via  `desktop.pedalboard.pedalboard("setOutputPortValue", ...)

Comment (falkTX) [inline mod/host.py]: The message needs to be saved so reloading the web page can re-send all the values.
The alternative would be to ask the plugin what its current state of a specific port + key is.

Comment (x42) [inline mod/host.py]: Nevermind you're right. This particular code is indeed not needed to emit atom msgs.

Comment (x42) [inline mod/host.py]: It's the plugin's DSP job to replay any messages (also in other LV2 hosts).

The common practice is for the GUI to send a message to the plugin when it is opened/loaded to query the state if it's needed.

**Pull request closed without merging**